json_serializable/lib/src/json_serializable_generator.dart

```diff
@@ -94,7 +94,7 @@ class JsonSerializableGenerator
     // Explicitly using `LinkedHashMap` – we want these ordered.
     var fields = new LinkedHashMap<String, FieldElement>.fromIterable(
         fieldsList,
-        key: (f) => (f as FieldElement).name);
+        key: (FieldElement f) => f.name);
```

Comment:
Should be `key: (Object f) => (f as FieldElement).name);`

or you get https://travis-ci.org/dart-lang/json_serializable/jobs/308340768#L588

`uses_dynamic_as_bottom`